Made dbname arg in sp_helpdb case-insensitive

RIC06X opened the pull request:
### Description

Previously sp_helpdb in babelfish treats dbname as case-sensitive whereas tsql treats the dbname argument as case-insensitive. This commit implemented a str_to_lower function for dbname and resolved the case-senstive issue.



### Issues Resolved

Made dbname arg in babelfish sp_helpdb case-insensitive

### Test Scenarios Covered ###
* **Use case based -**
test dbname with mixed letters of uppercase and lowercase
test dbname that end with trailing spaces

* **Negative test cases -**
test dbname with all lowercase letters

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

RIC06X committed:
Made dbname arg in sp_heapdb case-insensitive

Previously sp_helpdb in babelfish treats dbname as case-sensitive
whereas tsql treats the dbname argument as case-insensitive. This
commit implemented a str_to_lower function for dbname and resol-
ved the case-senstive issue.

Task: BABEL-3643
Signed-off-by: Yanjie Xu <yanjiex@amazon.com>

macwoody commented on contrib/babelfishpg_tsql/src/catalog.c:
```diff
@@ -344,6 +345,7 @@ babelfish_helpdb(PG_FUNCTION_ARGS)
 	if (PG_NARGS() > 0)
 	{
 		dbname = TextDatumGetCString(PG_GETARG_DATUM(0));
+		dbname = str_tolower(dbname, strlen(dbname), DEFAULT_COLLATION_OID);
```
if you overwrite this ptr, line 352's error message will show the lower case dbname that used for internal handling

macwoody commented on test/JDBC/expected/BABEL-3643.out:
```diff
@@ -0,0 +1,25 @@
+go
+~~START~~
+nvarchar#!#nvarchar#!#nvarchar#!#int#!#nvarchar#!#nvarchar#!#tinyint
+master#!#<NULL>#!#jdbc_user#!#1#!#2023-03-08 #!#<NULL>#!#<NULL>
```
sp_helpdb could not be tested like this, since it contains non-static information (timestamp). 
In general, all expected results should be consistent across run. 
Please refer to existing tests (I guess only tests the internal helper function and only select db name would be enough for your case)

RIC06X committed:
Fixed dbname not exist error msg

This commit fixed a issues that the database not exist error msg shows
all lowercased dbname instead of user typed one. We also added extra
corner case check that trailing space at the end of user typed dbname
will be removed and this allows sp_helpdb returns the correct database
information.

Task: BABEL-3643
Signed-off-by: Yanjie Xu <yanjiex@amazon.com>

lejaokri commented on contrib/babelfishpg_tsql/src/catalog.c:
```diff
@@ -344,14 +346,25 @@ babelfish_helpdb(PG_FUNCTION_ARGS)
+		dbname_lower = str_tolower(dbname, strlen(dbname), DEFAULT_COLLATION_OID);
+		/* Remove trailing spaces at the end of user typed dbname */
+		index = -1;
+		for (int i = 0; dbname_lower[i]!='\0'; i++)
+		{
+			if (dbname_lower[i]!=' ')
+			{
+				index = i;
+			}
+		}
```
nitpick
(1) is there already a trim() function in the code base we can use?
(2) do we need to pfree dbname_lower?

RIC06X committed:
Remove select items in sp_helpdb test

Since user could have different master db_size and the owner of the
database could be different if users have their own test configs, a
correct test result could show failure at the end. so we are removing
unnecessary select items in this test.

Task: BABEL-3643
Signed-off-by: Yanjie Xu <yanjiex@amazon.com>
